paseaf/ContainerSSH-honeypot
# Secure MinIO credentials; Fix dockerd on sacrificial VM; Change cAdvisor port

## Current steps
- [x] Write a bash script to generate credentials.
Abort if it's already generated.
- [x] MinIO run should import generated credentials
- [x] maybe update README (How user can find the generated credential for login)

## Description

This PR will
- close #24 
Added a script generate credentials to `/terraform/credentials.txt`.
It's then executed by `terraform apply` and pushed to GCP for configuration.
- make cAdvisor run from Docker on Sacrificial VM
Port change: `8080` -> `8088` (to be consistent with others)
- make `docker` command work on Sacrificial VM 
See Takeaways section below

## Testing the PR

Follow `/README.md` to
1. build Packer image
2. `terraform apply`
3. Open MinIO Console. You should be able to log in with generated credentials
4. Open Grafana. You should be able to see sacrificial-vm-cadvisor

## Takeaways
`docker` command didn't work on sacrificial VM because it was set to only listen to TCP port 2376.\
It works after making `dockerd` also listen to unix socket at `unix:///var/run/docker.sock`.
```diff
dockerd \
+  -H=unix:///var/run/docker.sock
    --tlsverify \
    --tlscacert=ca.pem \
    --tlscert=server-cert.pem \
    --tlskey=server-key.pem \
    -H=0.0.0.0:2376
```
Read more: [Docker Reference- dockerd](https://docs.docker.com/engine/reference/commandline/dockerd/)
  
## Q&A, Design Decisions
**Why not use some key management system like HashiCorp Vault?**
Because such systems are not trivial to set up, we don't need most of the benefits they provide.

**Credentials file naming**
Call it `credentials.txt` for users to open it more easily with text editor.
